PyRoute/DeltaDictionary.py

```diff
@@ -0,0 +1,368 @@
+            for subsector_name in self[sector_name]:
+                if subsector_name in subsectors:
+                    if sector_name not in overlap:
+                        overlap[sector_name] = list()
+                    overlap[sector_name].append(subsector_name)
+
+        new_dict = DeltaDictionary()
+        for sector_name in overlap:
+            new_dict[sector_name] = self[sector_name].subsector_subset(overlap[sector_name])
+
+        return new_dict
+
+    def sector_list(self):
+        result = list(self.keys())
+        result.sort()
+
+        return result
+
+    def subsector_list(self):
+        result = list()
+
+        for sector_name in self:
+            keys = self[sector_name].subsector_list()
+            result.extend(list(keys))
+
+        return result
+
+    @property
+    def lines(self):
+        result = list()
+        for sub_name in self.keys():
+            result.extend(self[sub_name].lines)
+
+        return result
+
+    def drop_lines(self, lines_to_drop):
+        foo = DeltaDictionary()
+        for sector_name in self:
+            if self[sector_name].skipped:
+                continue
+            else:
+                foo[sector_name] = self[sector_name].drop_lines(lines_to_drop)
+
+        return foo
+
+    def write_files(self, output_dir):
+        for sector_name in self:
+            self[sector_name].write_file(output_dir)
+
+    def skip_void_subsectors_if_half(self):
+        # if at least half of read-in subsectors are void, assume this is a follow-up on a previous reduction run
+        # and mark all void subsectors as skipped
+        void_subsec = 0
+        for secname in self:
+            void_subsec += self[secname].void_subsectors
+
+        if void_subsec * 2 >= 16 * len(self):
+            for secname in self:
+                self[secname].skip_void_subsectors()
+
+
+class SectorDictionary(dict):
+
+    def __init__(self, name, filename):
+        super().__init__()
+        self.name = name
+        self.filename = filename
+        self.position = None
+        self.allegiances = dict()
+        self.headers = list()
+
+    def update(self, __m, **kwargs):
+        for key in __m:
+            assert isinstance(__m[key], SubsectorDictionary), "Values must be SubsectorDictionary objects"
+        super().update(__m, **kwargs)
+
+    def __setitem__(self, item, value):
+        assert isinstance(value, SubsectorDictionary), "Values must be SubsectorDictionary objects"
+        super().__setitem__(item, value)
+
+    def __deepcopy__(self, memodict={}):
+        foo = SectorDictionary(self.name, self.filename)
+        foo.allegiances = copy.deepcopy(self.allegiances)
+        for alg in foo.allegiances:
+            foo.allegiances[alg].homeworlds = []
+            foo.allegiances[alg].stats.homeworlds = []
+            foo.allegiances[alg].stats.passengers = 0
+            foo.allegiances[alg].stats.trade = 0
+            foo.allegiances[alg].stats.tradeExt = 0
```

Comment:
As {Delta,Sector,Subsector}Dictionary objects are intended to serve as the _base_ data for the next pathfinding run, we need to explicitly zero the as-copied passenger/trade totals.  That way, those totals start from zero regardless of whether it's the 1st pathfinding pass in a debugging run, or the nth.

@tjoneslo , if you've got a better idea, let's hear it.